src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -118,6 +119,36 @@ object Instance {
     def apply(runSpec: RunSpec): Instance = Scheduled(runSpec, Id.forRunSpec(runSpec.id))
   }
 
+  object Provisioned {
+    /**
+      * Factory method for creating provisioned instance from Scheduled instance for apps
+      * @param scheduledInstance instance in a Scheduled state
+      * @return new instance in a provisioned state
+      */
+    def apply(scheduledInstance: Instance, agentInfo: AgentInfo, networkInfo: core.task.state.NetworkInfo, app: AppDefinition)(implicit clock: Clock): Instance = {
+      require(scheduledInstance.isScheduled, "Scheduled instance is required to create provisioned instance.")
+
+      val task = Task(
+        taskId = Task.Id.forInstanceId(scheduledInstance.instanceId, None),
```

Comment:
This is not correct for pods that have a container. I think you'll have to pass the container parameter as well. However, you can do that in the next step.

Reply:
this is just for apps, pods will have to have a different method because more stuff differ...

Reply:
as you can see, it accepts AppDefinition, not RunSpec

Reply:
Ah, you are right.